crawl/pom.py

```diff
@@ -537,20 +538,41 @@ def __init__(self, workspace, artifact_def, dependency, pom_template):
         super(DynamicPomGen, self).__init__(workspace, artifact_def, dependency)
         self.pom_content = workspace.pom_content
         self.pom_template = pom_template
+        self._workspace = workspace
 
     def gen(self, pomcontenttype):
         content = self.pom_template.replace("#{group_id}", self._artifact_def.group_id)
         content = content.replace("#{artifact_id}", self._artifact_def.artifact_id)
         version = self._artifact_def_version(pomcontenttype)
         content = content.replace("#{version}", version)
         content = self._handle_description(content, self.pom_content.description)
+
+        # Update the dependencies according to the overridden file
+        self.dependencies = self._update_overridden_deps(self.dependencies)
         if len(self.dependencies) == 0:
             content = self._remove_token(content, "#{dependencies}")
         else:
             content = content.replace(
                 "#{dependencies}", self._gen_dependencies(pomcontenttype))
         return content
 
+    def _update_overridden_deps(self, deps):
+        overrides_dict = self._workspace.name_to_override_dependencies
+        ext_deps = self._workspace.name_to_external_dependencies
+        output_deps = []
+        if overrides_dict == {}:
+            return deps
+        for dep in deps:
+            dep_str = dep
```

Comment:
This is confusing.  `dep_str` would make me think it is a `str` instance, but it isn't.